include/CLI/App.hpp

```diff
@@ -435,66 +435,28 @@ class App {
         return opt;
     }
 
-    /// Add option for vectors (no default)
+    /// Add option for vectors
     template <typename T>
     Option *add_option(std::string option_name,
                        std::vector<T> &variable, ///< The variable vector to set
                        std::string option_description = "",
-                       char delimiter = '\0') {
+                       bool defaulted = false) {
```

Comment:
The reason for having two copies of the defaulted method was to allow types that could not be converted to strings (or should not be converted to strings, such as an uninitialized int) - if you don't provide the defaulted argument, CLI11 does not try to make a default at all. However, for vectors, it might be combinable (since the uninitialized part is not a problem), but you'd need some template trickery to only convert from a string if it is supported.

Reply:
Should I add the overload back in?   I would argue if you are turning on defaulted an using an an uninitialized int that is a user issue.    But I could buy having a separate function that doesn't even have that code as you could have a variable type that doesn't have a << operator but does have a conversion from a string.

Reply:
We should add it back in for now. I am thinking it might be possible to have a template function that checks for the ability to convert and does the right thing in both cases. Yes, turning on defaulted for an uninitialized value is a user problem, though you have to be careful not to capture it as a string regardless, unfortunately (otherwise we could have a `->default()` option turn it on later)

Reply:
Is it valid to have a lambda function take an uninitialized value by capture? I think so, since we already do that to capture the normal value... That might actually mean `->default()` is possible...

Reply:
I added as as_string helper function in the transformer PR,  could convert to that for the default when/if that gets merged.  Might be possible to use some more template trickery to deal with some types that don't have a stream overload.

Reply:
I think capturing an uninitialized value would be undefined behavior if it is used before storing something to it.  Though for things like int and other trivial classes I think the effect is just a random non-nonsensical number being used, as long as it is not being used to index something else.

Reply:
> if it is used before

Yes, so:

```cpp
int x;
app.add_option("-x", x)->defaulted();
```

would be invalid (and hopefully might even be caught by the compiler), since you run the lambda function capturing the value of x when you run `defaulted()`. But this is a user issue and I think it would be okay. You could set the value of `x` in between the `add_option` and the `defaulted()`, and that would be fine. Suggestions for a better name (at least if this idea works) would be accepted...